CP77.CR2W/Modkit/UncookMLmask.cs

```diff
@@ -57,17 +63,32 @@ public static bool Uncook(CR2WFile cr2w, List<byte[]> buffers, FileInfo outfile,
 
             uint maskCount = blob.Header.NumLayers.val;
 
-            byte[] atlas = buffers[0];
-            uint[] tiles;
+            byte[] atlas;
+            var atlasRaw = new byte[atlasWidth * atlasHeight];
+            var atlasBuffer = cr2w.Buffers[0].Buffer;
+            cr2wStream.Seek(atlasBuffer.offset, SeekOrigin.Begin);
+            using (var ms = new MemoryStream())
+            {
+                cr2wStream.DecompressAndCopySegment(ms, atlasBuffer.diskSize, atlasBuffer.memSize);
+                atlas = ms.ToArray();
+            }
+
 
             //Read tilesdata buffer into appropriate variable type
-            var b = buffers[1];
-            tiles = new uint[b.Length / 4];
-            for (int i = 0, j = 0; i < b.Length; i += 4, j++)
+            byte[] tile;
+            var tileBuffer = cr2w.Buffers[0].Buffer;
+            cr2wStream.Seek(tileBuffer.offset, SeekOrigin.Begin);
+            using (var ms = new MemoryStream())
+            {
+                cr2wStream.DecompressAndCopySegment(ms, tileBuffer.diskSize, tileBuffer.memSize);
+                tile = ms.ToArray();
```

Comment:
Is `ToArray` necessary? Couldn't this be just a stream and you read from it instead of copy all bytes just to do one thing?